metricflow/model/semantics/semantic_containers.py

```diff
@@ -248,6 +248,13 @@ def get_identifier_references(self) -> List[IdentifierReference]:  # noqa: D
         return list(self._identifier_ref_to_entity.keys())
 
     # DSC interface
+    @staticmethod
+    def find_primary_time_dimension(data_source: DataSource) -> TimeDimensionReference:  # noqa: D
+        for dimension in data_source.dimensions:
+            if dimension.type == DimensionType.TIME and dimension.type_params and dimension.type_params.is_primary:
```

Comment:
See this goes to the DataSource, not the source node set, which is why I was wondering if this accounts for metric_time usage on measures with a different agg_time_dimension.

It's possible for a data source to have no primary time dimension, I think, or it used to be. Regardless it's possible for data_source.primary_time to never be referenced anywhere.

Reply:
now that I think about it, you're right. This doesn't cover the case of measures with different agg_time_dimension. With more thought on that, I actually realized we didn't need to do any of the finding primary time dimension stuff. Since the source node splits the nodes into measures with the same agg_time_dimension. I can just track the measure's agg_time_dimension and compare that to the non_additive_dimension_spec to check whether `metric_time` is a match or not.  So I ended up wiping all the code in favour of that. https://github.com/transform-data/metricflow/pull/253/commits/da5ce02dc2b0922e3a90f505e450c9abeea8f24b